src/parsers/js/transformers/eslint/index.js

```diff
@@ -0,0 +1,66 @@
+import compileModule from '../../../utils/compileModule';
+import pkg from 'eslint/package.json';
+
+const ID = 'eslint1';
+
+export default {
+  id: ID,
+  displayName: ID,
+  version: pkg.version,
+  homepage: pkg.homepage,
+
+  defaultParserID: 'acorn-to-esprima',
+
+  loadTransformer(callback) {
+    require(
+      [
+        // Explicitly require just the stuff we care about to avoid loading
+        // RuleTester and CLIEngine, which are unnecessary and bloat out the
+        // package size.
+        'eslint/lib/eslint',
+        'eslint/lib/util/source-code',
+        'eslint/lib/rules',
+        'babel-core',
+        'babel-eslint'
+      ],
+      (eslint, eslintSourceCode, eslintRules, babel, babelESLint) => {
+        callback({ eslint, eslintSourceCode, eslintRules, babel, babelESLint });
+      }
+    );
+  },
+
+  transform({ eslint, eslintSourceCode, eslintRules, babel, babelESLint }, transformCode, code) {
+    // Compile the transform code and install it as an ESLint rule. The rule
+    // name doesn't really matter here, so we'll just use a hard-coded name.
+    const rule = compileModule(
+      babel.transform(transformCode).code
+    );
+    eslintRules.define('astExplorerRule', rule);
+
+    // Run the ESLint rule on the AST of the provided code.
+    // Reference: http://eslint.org/docs/developer-guide/nodejs-api
+    const ast = babelESLint.parseNoPatch(code);
+    const sourceCode = new eslintSourceCode(code, ast);
+    const results = eslint.verify(sourceCode, {
+      rules: {
+        astExplorerRule: 2
+      }
+    });
+    return formatResults(results);
+  },
+};
+
+function formatResults(results) {
+  return results.length === 0
+    ? 'Lint rule not fired.'
+    : results.map(formatResult).join('').trim();
+}
+
+function formatResult(result) {
+  var pointer = '-'.repeat(result.column - 1) + '^';
+  return `
+\u2022 ${result.message} (at ${result.line}:${result.column})
```

Comment:
Is there a way to disable syntax highlighting in the output pane? I'd
rather just totally disable syntax highlighting for ESLint results as it
doesn't really make sense.

Sent from my phone.
On Mar 22, 2016 8:55 AM, "Ingvar Stepanyan" <notifications@github.com>
wrote:

> In src/parsers/js/transformers/eslint/index.js
> <https://github.com/fkling/astexplorer/pull/104#discussion_r57012540>:
>
> > +      }
> > +    });
> > +    return formatResults(results);
> > +  },
> > +};
> > +
> > +function formatResults(results) {
> > +  return results.length === 0
> > +    ? 'Lint rule not fired.'
> > +    : results.map(formatResult).join('').trim();
> > +}
> > +
> > +function formatResult(result) {
> > +  var pointer = '-'.repeat(result.column - 1) + '^';
> > +  return `
> > +\u2022 ${result.message} (at ${result.line}:${result.column})
>
> With wrapping into ' you need to care about internal possible ' in error
> messages. I would rather go with // as it can be terminated only by line
> terminator :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/fkling/astexplorer/pull/104/files/eb0db992148047d55addcf1086fd0de92f3b6c92#r57012540>
>